Description: Allow traditionally delimited rules.
 A rule of unlimited validity must be ot the form
 `s/this/that', without a final delimiting dash.
 This will confuse traditionalists like myself,
 expecting 's/this/that/' to deliver the same effect.
 .
 Thus make a final dash optional, like common sed-syntax.
Author: Mats Erik Andersson <debian@gisladisker.se>
Forwarded: yes
Last-Update: 2011-07-19

diff -Naur netsed-1.00b.debian/netsed.c netsed-1.00b/netsed.c
--- netsed-1.00b.debian/netsed.c
+++ netsed-1.00b/netsed.c
@@ -564,7 +564,8 @@
     if (cs) { *cs=0; cs++; }
     rule[rules].forig=fs;
     rule[rules].torig=ts;
-    if (cs) rule_live[rules]=atoi(cs); else rule_live[rules]=-1;
+    if (cs && *cs) /* Only non-trivial quantifiers count. */
+      rule_live[rules]=atoi(cs); else rule_live[rules]=-1;
     shrink_to_binary(&rule[rules]);
 //    printf("DEBUG: (%s) (%s)\n",rule[rules].from,rule[rules].to);
     rules++;
